### ISSUE TITLE
Plugin-Management: Finetune modules loaded and displayed in module manager

### ISSUE THREAD

**Issue by ursjoss** [thinkingrock-gtd/tr-pc#57]
The list of modules loaded in ThinkingRock-3.7.0 and that in the current development version differs considerably.

### Modules loaded (according to log files)

<details><summary>Modules loaded in 3.7.0 (sorted w/o version indication)</summary>
<pre>
	activation
	apache.commons.email
	au.com.trgtd.tr.appl
	au.com.trgtd.tr.archive
	au.com.trgtd.tr.cal
	au.com.trgtd.tr.calendar
	au.com.trgtd.tr.calendar.ical4j
	au.com.trgtd.tr.calendar.ical4j.impl
	au.com.trgtd.tr.data
	au.com.trgtd.tr.data.recent
	au.com.trgtd.tr.datastore
	au.com.trgtd.tr.datastore.xstream
	au.com.trgtd.tr.email
	au.com.trgtd.tr.export
	au.com.trgtd.tr.export.actions
	au.com.trgtd.tr.export.data
	au.com.trgtd.tr.export.references
	au.com.trgtd.tr.export.someday
	au.com.trgtd.tr.extract
	au.com.trgtd.tr.find
	au.com.trgtd.tr.imports
	au.com.trgtd.tr.imports.thoughts
	au.com.trgtd.tr.l10n.de_DE
	au.com.trgtd.tr.l10n.en_US
	au.com.trgtd.tr.l10n.es_ES
	au.com.trgtd.tr.l10n.fr_FR
	au.com.trgtd.tr.prefs.actions
	au.com.trgtd.tr.prefs.data
	au.com.trgtd.tr.prefs.dates
	au.com.trgtd.tr.prefs.projects
	au.com.trgtd.tr.prefs.recurrence
	au.com.trgtd.tr.prefs.ui
	au.com.trgtd.tr.report.project.detail
	au.com.trgtd.tr.report.project.outline
	au.com.trgtd.tr.report.projects.future
	au.com.trgtd.tr.report.sa
	au.com.trgtd.tr.reports.actions.delegated
	au.com.trgtd.tr.reports.actions.doasap
	au.com.trgtd.tr.reports.actions.scheduled
	au.com.trgtd.tr.reports.done
	au.com.trgtd.tr.reports.pocketmod
	au.com.trgtd.tr.reports.reference
	au.com.trgtd.tr.reports.someday
	au.com.trgtd.tr.reports.weekly
	au.com.trgtd.tr.resource
	au.com.trgtd.tr.runtime
	au.com.trgtd.tr.services
	au.com.trgtd.tr.swing
	au.com.trgtd.tr.sync.device
	au.com.trgtd.tr.sync.iphone
	au.com.trgtd.tr.task.activation
	au.com.trgtd.tr.task.messages
	au.com.trgtd.tr.task.recurrence
	au.com.trgtd.tr.util
	au.com.trgtd.tr.view
	au.com.trgtd.tr.view.actn
	au.com.trgtd.tr.view.actns
	au.com.trgtd.tr.view.calendar
	au.com.trgtd.tr.view.collect
	au.com.trgtd.tr.view.contexts
	au.com.trgtd.tr.view.criteria
	au.com.trgtd.tr.view.delegates
	au.com.trgtd.tr.view.filters
	au.com.trgtd.tr.view.goals
	au.com.trgtd.tr.view.overview
	au.com.trgtd.tr.view.process
	au.com.trgtd.tr.view.project
	au.com.trgtd.tr.view.projects
	au.com.trgtd.tr.view.reference
	au.com.trgtd.tr.view.someday
	au.com.trgtd.tr.view.topics
	ca.odell.glazedlists
	commons.lang
	commons.logging
	com.thoughtworks.xstream
	fop
	ical4j
INFO
	jasper
	javamail
	jaxb
	miglayout
	minimal_json
	org.jdesktop.layout/1
	org.netbeans.api.annotations.common/1
	org.netbeans.api.intent
	org.netbeans.api.io
	org.netbeans.api.progress/1
	org.netbeans.api.progress.compat8
	org.netbeans.api.progress.nb
	org.netbeans.api.templates
	org.netbeans.bootstrap/1
	org.netbeans.core/2
	org.netbeans.core.execution/1
	org.netbeans.core.multitabs/1
	org.netbeans.core.nativeaccess/1
	org.netbeans.core.network
	org.netbeans.core.output2/1
	org.netbeans.core.startup/1
	org.netbeans.core.startup.base
	org.netbeans.core.ui/1
	org.netbeans.core.windows/2
	org.netbeans.libs.asm
	org.netbeans.libs.jna/1
	org.netbeans.libs.jna.platform/1
	org.netbeans.modules.autoupdate.services
	org.netbeans.modules.autoupdate.ui
	org.netbeans.modules.editor.mimelookup/1
	org.netbeans.modules.extbrowser/1
	org.netbeans.modules.extexecution/2
	org.netbeans.modules.extexecution.base/2
	org.netbeans.modules.javahelp/1
	org.netbeans.modules.keyring
	org.netbeans.modules.masterfs/2
	org.netbeans.modules.masterfs.linux
	org.netbeans.modules.masterfs.ui
	org.netbeans.modules.options.api/1
	org.netbeans.modules.options.keymap
	org.netbeans.modules.progress.ui
	org.netbeans.modules.queries/1
	org.netbeans.modules.sampler
	org.netbeans.modules.settings/1
	org.netbeans.spi.quicksearch
	org.netbeans.swing.outline
	org.netbeans.swing.plaf
	org.netbeans.swing.tabcontrol
	org.openide.actions
	org.openide.awt
	org.openide.dialogs
	org.openide.execution
	org.openide.execution.compat8
	org.openide.explorer
	org.openide.filesystems
	org.openide.filesystems.compat8
	org.openide.filesystems.nb
	org.openide.io
	org.openide.loaders
	org.openide.modules
	org.openide.nodes
	org.openide.text
	org.openide.util
	org.openide.util.lookup
	org.openide.util.ui
	org.openide.windows
	swingx
	tr.extract.reports
	tr.extract.reports.projectdetails
	tr.extract.reports.projectoutline
	tr.model
	xalan
</pre>
</details>

<details><summary>Modules loaded in the developement version (sorted, w/o version indication)</summary>
<pre>
	activation
	apache.commons.email
	au.com.trgtd.tr.appl
	au.com.trgtd.tr.archive
	au.com.trgtd.tr.cal
	au.com.trgtd.tr.calendar
	au.com.trgtd.tr.calendar.ical4j
	au.com.trgtd.tr.calendar.ical4j.impl
	au.com.trgtd.tr.data
	au.com.trgtd.tr.data.recent
	au.com.trgtd.tr.datastore
	au.com.trgtd.tr.datastore.xstream
	au.com.trgtd.tr.email
	au.com.trgtd.tr.export
	au.com.trgtd.tr.export.actions
	au.com.trgtd.tr.export.data
	au.com.trgtd.tr.export.references
	au.com.trgtd.tr.export.someday
	au.com.trgtd.tr.extract
	au.com.trgtd.tr.find
	au.com.trgtd.tr.imports
	au.com.trgtd.tr.imports.thoughts
	au.com.trgtd.tr.l10n.de_DE
	au.com.trgtd.tr.l10n.en_US
	au.com.trgtd.tr.l10n.es_ES
	au.com.trgtd.tr.l10n.fr_FR
	au.com.trgtd.tr.prefs.actions
	au.com.trgtd.tr.prefs.data
	au.com.trgtd.tr.prefs.dates
	au.com.trgtd.tr.prefs.projects
	au.com.trgtd.tr.prefs.recurrence
	au.com.trgtd.tr.prefs.ui
	au.com.trgtd.tr.report.project.detail
	au.com.trgtd.tr.report.project.outline
	au.com.trgtd.tr.report.projects.future
	au.com.trgtd.tr.report.sa
	au.com.trgtd.tr.reports.actions.delegated
	au.com.trgtd.tr.reports.actions.doasap
	au.com.trgtd.tr.reports.actions.scheduled
	au.com.trgtd.tr.reports.done
	au.com.trgtd.tr.reports.pocketmod
	au.com.trgtd.tr.reports.reference
	au.com.trgtd.tr.reports.someday
	au.com.trgtd.tr.reports.weekly
	au.com.trgtd.tr.resource
	au.com.trgtd.tr.runtime
	au.com.trgtd.tr.services
	au.com.trgtd.tr.swing
	au.com.trgtd.tr.sync.device
	au.com.trgtd.tr.sync.iphone
	au.com.trgtd.tr.task.activation
	au.com.trgtd.tr.task.messages
	au.com.trgtd.tr.task.recurrence
	au.com.trgtd.tr.util
	au.com.trgtd.tr.view
	au.com.trgtd.tr.view.actn
	au.com.trgtd.tr.view.actns
	au.com.trgtd.tr.view.calendar
	au.com.trgtd.tr.view.collect
	au.com.trgtd.tr.view.contexts
	au.com.trgtd.tr.view.criteria
	au.com.trgtd.tr.view.delegates
	au.com.trgtd.tr.view.filters
	au.com.trgtd.tr.view.goals
	au.com.trgtd.tr.view.overview
	au.com.trgtd.tr.view.process
	au.com.trgtd.tr.view.project
	au.com.trgtd.tr.view.projects
	au.com.trgtd.tr.view.reference
	au.com.trgtd.tr.view.someday
	au.com.trgtd.tr.view.topics
	bcpg
	bcprov
	ca.odell.glazedlists
	com.googlecode.javaewah.JavaEWAH
	com.google.gson
	com.google.guava
	com.jcraft.jsch
	com.jcraft.jzlib
	commons.lang
	commons.lang3
	commons.logging
	commons.text
	com.thoughtworks.xstream
	fop
	ical4j
INFO
	jasper
	javamail
	jaxb
	libs.c.kohlschutter.junixsocket/2
	miglayout
	minimal_json
	org.apache.commons.codec
	org.apache.commons.httpclient
	org.apache.commons.lang
	org.apache.commons.logging
	org.apache.ws.commons.util
	org.apache.xml.resolver
	org.apache.xmlrpc
	org.eclipse.core.contenttype
	org.eclipse.core.jobs
	org.eclipse.core.net
	org.eclipse.core.runtime
	org.eclipse.core.runtime.compatibility.auth
	org.eclipse.equinox.app
	org.eclipse.equinox.common
	org.eclipse.equinox.preferences
	org.eclipse.equinox.registry
	org.eclipse.equinox.security
	org.eclipse.jgit
	org.eclipse.mylyn.bugzilla.core
	org.eclipse.mylyn.commons.core
	org.eclipse.mylyn.commons.net
	org.eclipse.mylyn.commons.repositories.core
	org.eclipse.mylyn.commons.xmlrpc
	org.eclipse.mylyn.tasks.core
	org.eclipse.mylyn.wikitext.core
	org.netbeans.api.annotations.common/1
	org.netbeans.api.debugger/1
	org.netbeans.api.intent
	org.netbeans.api.io
	org.netbeans.api.java.classpath/1
	org.netbeans.api.lsp/1
	org.netbeans.api.progress/1
	org.netbeans.api.progress.compat8
	org.netbeans.api.progress.nb
	org.netbeans.api.scripting
	org.netbeans.api.search
	org.netbeans.api.templates
	org.netbeans.api.visual
	org.netbeans.api.xml/1
	org.netbeans.api.xml.ui/1
	org.netbeans.bootstrap/1
	org.netbeans.core/2
	org.netbeans.core.browser/1
	org.netbeans.core.execution/1
	org.netbeans.core.ide/1
	org.netbeans.core.io.ui/1
	org.netbeans.core.multitabs/1
	org.netbeans.core.multitabs.project
	org.netbeans.core.multiview/1
	org.netbeans.core.nativeaccess/1
	org.netbeans.core.netigso
	org.netbeans.core.network
	org.netbeans.core.output2/1
	org.netbeans.core.startup/1
	org.netbeans.core.startup.base
	org.netbeans.core.ui/1
	org.netbeans.core.windows/2
	org.netbeans.libs.antlr3.runtime
	org.netbeans.libs.antlr4.runtime
	org.netbeans.libs.asm
	org.netbeans.libs.batik.read
	org.netbeans.libs.commons_compress/0
	org.netbeans.libs.commons_net/2
	org.netbeans.libs.felix
	org.netbeans.libs.flatlaf/1
	org.netbeans.libs.flexmark
	org.netbeans.libs.freemarker/1
	org.netbeans.libs.git/1
	org.netbeans.libs.graalsdk
	org.netbeans.libs.ini4j/1
	org.netbeans.libs.jcodings/1
	org.netbeans.libs.jna/2
	org.netbeans.libs.jna.platform/2
	org.netbeans.libs.jsch.agentproxy/2
	org.netbeans.libs.json_simple/1
	org.netbeans.libs.lucene/3
	org.netbeans.libs.osgi
	org.netbeans.libs.snakeyaml_engine/2
	org.netbeans.libs.svnClientAdapter/1
	org.netbeans.libs.svnClientAdapter.javahl/1
	org.netbeans.libs.tomlj
	org.netbeans.libs.truffleapi
	org.netbeans.libs.xerces/1
	org.netbeans.lib.terminalemulator
	org.netbeans.modules.autoupdate.cli
	org.netbeans.modules.autoupdate.services
	org.netbeans.modules.autoupdate.ui
	org.netbeans.modules.bugtracking
	org.netbeans.modules.bugtracking.bridge
	org.netbeans.modules.bugtracking.commons
	org.netbeans.modules.bugzilla
	org.netbeans.modules.core.kit
	org.netbeans.modules.csl.api/2
	org.netbeans.modules.csl.types/1
	org.netbeans.modules.css.editor/1
	org.netbeans.modules.css.lib/1
	org.netbeans.modules.css.model
	org.netbeans.modules.css.prep
	org.netbeans.modules.css.visual/3
	org.netbeans.modules.db/1
	org.netbeans.modules.dbapi
	org.netbeans.modules.db.core
	org.netbeans.modules.db.dataview
	org.netbeans.modules.db.drivers
	org.netbeans.modules.db.kit
	org.netbeans.modules.db.metadata.model/1
	org.netbeans.modules.db.mysql
	org.netbeans.modules.db.sql.editor
	org.netbeans.modules.db.sql.visualeditor/1
	org.netbeans.modules.defaults/1
	org.netbeans.modules.derby
	org.netbeans.modules.diff/1
	org.netbeans.modules.dlight.nativeexecution
	org.netbeans.modules.dlight.nativeexecution.nb
	org.netbeans.modules.dlight.terminal
	org.netbeans.modules.docker.api/0
	org.netbeans.modules.docker.editor/0
	org.netbeans.modules.docker.ui/0
	org.netbeans.modules.editor/3
	org.netbeans.modules.editor.actions/1
	org.netbeans.modules.editor.autosave/1
	org.netbeans.modules.editor.bookmarks/1
	org.netbeans.modules.editor.bracesmatching/0
	org.netbeans.modules.editor.breadcrumbs/0
	org.netbeans.modules.editor.codetemplates/1
	org.netbeans.modules.editor.completion/1
	org.netbeans.modules.editor.deprecated.pre65formatting/0
	org.netbeans.modules.editor.document
	org.netbeans.modules.editor.errorstripe/2
	org.netbeans.modules.editor.errorstripe.api/1
	org.netbeans.modules.editor.fold/1
	org.netbeans.modules.editor.fold.nbui
	org.netbeans.modules.editor.global.format/1
	org.netbeans.modules.editor.guards/1
	org.netbeans.modules.editor.indent/2
	org.netbeans.modules.editor.indent.project/0
	org.netbeans.modules.editor.indent.support
	org.netbeans.modules.editor.kit
	org.netbeans.modules.editor.lib2/1
	org.netbeans.modules.editor.lib/3
	org.netbeans.modules.editor.macros/0
	org.netbeans.modules.editor.mimelookup/1
	org.netbeans.modules.editor.mimelookup.impl/1
	org.netbeans.modules.editor.plain/2
	org.netbeans.modules.editor.plain.lib/1
	org.netbeans.modules.editor.search
	org.netbeans.modules.editor.settings/1
	org.netbeans.modules.editor.settings.lib
	org.netbeans.modules.editor.settings.storage/1
	org.netbeans.modules.editor.structure/1
	org.netbeans.modules.editor.tools.storage
	org.netbeans.modules.editor.util/1
	org.netbeans.modules.extbrowser/1
	org.netbeans.modules.extexecution/2
	org.netbeans.modules.extexecution.base/2
	org.netbeans.modules.extexecution.impl
	org.netbeans.modules.extexecution.process
	org.netbeans.modules.extexecution.process.jdk9
	org.netbeans.modules.favorites/1
	org.netbeans.modules.git
	org.netbeans.modules.gsf.testrunner/2
	org.netbeans.modules.gsf.testrunner.ui
	org.netbeans.modules.html/1
	org.netbeans.modules.html.custom
	org.netbeans.modules.html.editor/2
	org.netbeans.modules.html.editor.lib/3
	org.netbeans.modules.html.indexing
	org.netbeans.modules.html.lexer/1
	org.netbeans.modules.html.parser/1
	org.netbeans.modules.html.validation/1
	org.netbeans.modules.httpserver/2
	org.netbeans.modules.hudson
	org.netbeans.modules.hudson.git
	org.netbeans.modules.hudson.mercurial
	org.netbeans.modules.hudson.subversion
	org.netbeans.modules.hudson.tasklist
	org.netbeans.modules.hudson.ui
	org.netbeans.modules.ide.kit
	org.netbeans.modules.image/1
	org.netbeans.modules.janitor
	org.netbeans.modules.javahelp/1
	org.netbeans.modules.jumpto/1
	org.netbeans.modules.keyring
	org.netbeans.modules.keyring.fallback
	org.netbeans.modules.keyring.impl
	org.netbeans.modules.languages.diff
	org.netbeans.modules.languages.manifest
	org.netbeans.modules.languages.toml
	org.netbeans.modules.languages.yaml
	org.netbeans.modules.lexer/2
	org.netbeans.modules.lexer.nbbridge/1
	org.netbeans.modules.localhistory
	org.netbeans.modules.localtasks
	org.netbeans.modules.lsp.client/0
	org.netbeans.modules.markdown
	org.netbeans.modules.masterfs/2
	org.netbeans.modules.masterfs.linux
	org.netbeans.modules.masterfs.nio2
	org.netbeans.modules.masterfs.ui
	org.netbeans.modules.mercurial
	org.netbeans.modules.mylyn.util
	org.netbeans.modules.nativeimage.api/0
	org.netbeans.modules.netbinox
	org.netbeans.modules.notifications
	org.netbeans.modules.options.api/1
	org.netbeans.modules.options.editor/1
	org.netbeans.modules.options.keymap
	org.netbeans.modules.parsing.api/1
	org.netbeans.modules.parsing.indexing
	org.netbeans.modules.parsing.lucene/2
	org.netbeans.modules.parsing.nb
	org.netbeans.modules.parsing.ui
	org.netbeans.modules.print
	org.netbeans.modules.print.editor
	org.netbeans.modules.progress.ui
	org.netbeans.modules.project.ant/1
	org.netbeans.modules.projectapi/1
	org.netbeans.modules.projectapi.nb
	org.netbeans.modules.project.indexingbridge
	org.netbeans.modules.project.libraries/1
	org.netbeans.modules.project.libraries.ui/1
	org.netbeans.modules.project.spi.intern
	org.netbeans.modules.project.spi.intern.impl
	org.netbeans.modules.projectui
	org.netbeans.modules.projectuiapi/1
	org.netbeans.modules.projectuiapi.base/1
	org.netbeans.modules.properties/1
	org.netbeans.modules.properties.syntax/1
	org.netbeans.modules.queries/1
	org.netbeans.modules.refactoring.api
	org.netbeans.modules.sampler
	org.netbeans.modules.selenium2
	org.netbeans.modules.selenium2.server
	org.netbeans.modules.sendopts/2
	org.netbeans.modules.server/0
	org.netbeans.modules.servletapi/1
	org.netbeans.modules.settings/1
	org.netbeans.modules.spellchecker
	org.netbeans.modules.spellchecker.apimodule
	org.netbeans.modules.spellchecker.bindings.htmlxml
	org.netbeans.modules.spellchecker.bindings.properties
	org.netbeans.modules.spellchecker.dictionary_en
	org.netbeans.modules.spellchecker.kit
	org.netbeans.modules.subversion
	org.netbeans.modules.tasklist.kit
	org.netbeans.modules.tasklist.projectint/1
	org.netbeans.modules.tasklist.todo/1
	org.netbeans.modules.tasklist.ui/1
	org.netbeans.modules.team.commons
	org.netbeans.modules.team.ide
	org.netbeans.modules.templates/1
	org.netbeans.modules.terminal
	org.netbeans.modules.terminal.nb
	org.netbeans.modules.textmate.lexer/0
	org.netbeans.modules.usersguide/1
	org.netbeans.modules.utilities/1
	org.netbeans.modules.utilities.project/1
	org.netbeans.modules.versioning/1
	org.netbeans.modules.versioning.core/1
	org.netbeans.modules.versioning.indexingbridge/0
	org.netbeans.modules.versioning.masterfs
	org.netbeans.modules.versioning.system.cvss.installer
	org.netbeans.modules.versioning.ui/1
	org.netbeans.modules.versioning.util
	org.netbeans.modules.web.browser.api
	org.netbeans.modules.web.common
	org.netbeans.modules.web.common.ui
	org.netbeans.modules.web.indent
	org.netbeans.modules.xml
	org.netbeans.modules.xml.axi
	org.netbeans.modules.xml.catalog/2
	org.netbeans.modules.xml.catalog.ui
	org.netbeans.modules.xml.core/2
	org.netbeans.modules.xml.lexer
	org.netbeans.modules.xml.retriever/1
	org.netbeans.modules.xml.schema.completion
	org.netbeans.modules.xml.schema.model/1
	org.netbeans.modules.xml.tax/2
	org.netbeans.modules.xml.text/2
	org.netbeans.modules.xml.tools/2
	org.netbeans.modules.xml.xam/1
	org.netbeans.modules.xml.xdm/1
	org.netbeans.modules.xsl/1
	org.netbeans.spi.editor.hints/0
	org.netbeans.spi.editor.hints.projects
	org.netbeans.spi.navigator/1
	org.netbeans.spi.palette/1
	org.netbeans.spi.quicksearch
	org.netbeans.spi.tasklist/1
	org.netbeans.swing.dirchooser
	org.netbeans.swing.laf.dark/1
	org.netbeans.swing.laf.flatlaf
	org.netbeans.swing.outline
	org.netbeans.swing.plaf
	org.netbeans.swing.tabcontrol
	org.openide.actions
	org.openide.awt
	org.openide.dialogs
	org.openide.execution
	org.openide.execution.compat8
	org.openide.explorer
	org.openide.filesystems
	org.openide.filesystems.compat8
	org.openide.filesystems.nb
	org.openide.io
	org.openide.loaders
	org.openide.modules
	org.openide.nodes
	org.openide.text
	org.openide.util
	org.openide.util.lookup
	org.openide.util.ui
	org.openide.util.ui.svg
	org.openide.windows
	slf4j.api
	slf4j.jdk14
	swingx
	tr.extract.reports
	tr.extract.reports.projectdetails
	tr.extract.reports.projectoutline
	tr.model
	xalan
</pre>
</details>

### Modules not loaded (according to log files)

<details><summary>Modules not loaded in the development version</summary>
<pre>
       org.jdesktop.layout/1
</pre>
</details>

<details><summary>Modules not loaded in the 3.7.0</summary>
<pre>
 	bcpg
 	bcprov
 	com.googlecode.javaewah.JavaEWAH
 	com.google.gson
 	com.google.guava
 	com.jcraft.jsch
 	com.jcraft.jzlib
 	commons.lang3
 	commons.text
 	libs.c.kohlschutter.junixsocket/2
 	org.apache.commons.codec
 	org.apache.commons.httpclient
 	org.apache.commons.lang
 	org.apache.commons.logging
 	org.apache.ws.commons.util
 	org.apache.xml.resolver
 	org.apache.xmlrpc
 	org.eclipse.core.contenttype
 	org.eclipse.core.jobs
 	org.eclipse.core.net
 	org.eclipse.core.runtime
 	org.eclipse.core.runtime.compatibility.auth
 	org.eclipse.equinox.app
 	org.eclipse.equinox.common
 	org.eclipse.equinox.preferences
 	org.eclipse.equinox.registry
 	org.eclipse.equinox.security
 	org.eclipse.jgit
 	org.eclipse.mylyn.bugzilla.core
 	org.eclipse.mylyn.commons.core
 	org.eclipse.mylyn.commons.net
 	org.eclipse.mylyn.commons.repositories.core
 	org.eclipse.mylyn.commons.xmlrpc
 	org.eclipse.mylyn.tasks.core
 	org.eclipse.mylyn.wikitext.core
 	org.netbeans.api.debugger/1
 	org.netbeans.api.java.classpath/1
 	org.netbeans.api.lsp/1
 	org.netbeans.api.scripting
 	org.netbeans.api.search
 	org.netbeans.api.visual
 	org.netbeans.api.xml/1
 	org.netbeans.api.xml.ui/1
 	org.netbeans.core.browser/1
 	org.netbeans.core.ide/1
 	org.netbeans.core.io.ui/1
 	org.netbeans.core.multitabs.project
 	org.netbeans.core.multiview/1
 	org.netbeans.core.netigso
 	org.netbeans.libs.antlr3.runtime
 	org.netbeans.libs.antlr4.runtime
 	org.netbeans.libs.batik.read
 	org.netbeans.libs.commons_compress/0
 	org.netbeans.libs.commons_net/2
 	org.netbeans.libs.felix
 	org.netbeans.libs.flatlaf/1
 	org.netbeans.libs.flexmark
 	org.netbeans.libs.freemarker/1
 	org.netbeans.libs.git/1
 	org.netbeans.libs.graalsdk
 	org.netbeans.libs.ini4j/1
 	org.netbeans.libs.jcodings/1
 	org.netbeans.libs.jsch.agentproxy/2
 	org.netbeans.libs.json_simple/1
 	org.netbeans.libs.lucene/3
 	org.netbeans.libs.osgi
 	org.netbeans.libs.snakeyaml_engine/2
 	org.netbeans.libs.svnClientAdapter/1
 	org.netbeans.libs.svnClientAdapter.javahl/1
 	org.netbeans.libs.tomlj
 	org.netbeans.libs.truffleapi
 	org.netbeans.libs.xerces/1
 	org.netbeans.lib.terminalemulator
 	org.netbeans.modules.autoupdate.cli
 	org.netbeans.modules.bugtracking
 	org.netbeans.modules.bugtracking.bridge
 	org.netbeans.modules.bugtracking.commons
 	org.netbeans.modules.bugzilla
 	org.netbeans.modules.core.kit
 	org.netbeans.modules.csl.api/2
 	org.netbeans.modules.csl.types/1
 	org.netbeans.modules.css.editor/1
 	org.netbeans.modules.css.lib/1
 	org.netbeans.modules.css.model
 	org.netbeans.modules.css.prep
 	org.netbeans.modules.css.visual/3
 	org.netbeans.modules.db/1
 	org.netbeans.modules.dbapi
 	org.netbeans.modules.db.core
 	org.netbeans.modules.db.dataview
 	org.netbeans.modules.db.drivers
 	org.netbeans.modules.db.kit
 	org.netbeans.modules.db.metadata.model/1
 	org.netbeans.modules.db.mysql
 	org.netbeans.modules.db.sql.editor
 	org.netbeans.modules.db.sql.visualeditor/1
 	org.netbeans.modules.defaults/1
 	org.netbeans.modules.derby
 	org.netbeans.modules.diff/1
 	org.netbeans.modules.dlight.nativeexecution
 	org.netbeans.modules.dlight.nativeexecution.nb
 	org.netbeans.modules.dlight.terminal
 	org.netbeans.modules.docker.api/0
 	org.netbeans.modules.docker.editor/0
 	org.netbeans.modules.docker.ui/0
 	org.netbeans.modules.editor/3
 	org.netbeans.modules.editor.actions/1
 	org.netbeans.modules.editor.autosave/1
 	org.netbeans.modules.editor.bookmarks/1
 	org.netbeans.modules.editor.bracesmatching/0
 	org.netbeans.modules.editor.breadcrumbs/0
 	org.netbeans.modules.editor.codetemplates/1
 	org.netbeans.modules.editor.completion/1
 	org.netbeans.modules.editor.deprecated.pre65formatting/0
 	org.netbeans.modules.editor.document
 	org.netbeans.modules.editor.errorstripe/2
 	org.netbeans.modules.editor.errorstripe.api/1
 	org.netbeans.modules.editor.fold/1
 	org.netbeans.modules.editor.fold.nbui
 	org.netbeans.modules.editor.global.format/1
 	org.netbeans.modules.editor.guards/1
 	org.netbeans.modules.editor.indent/2
 	org.netbeans.modules.editor.indent.project/0
 	org.netbeans.modules.editor.indent.support
 	org.netbeans.modules.editor.kit
 	org.netbeans.modules.editor.lib2/1
 	org.netbeans.modules.editor.lib/3
 	org.netbeans.modules.editor.macros/0
 	org.netbeans.modules.editor.mimelookup.impl/1
 	org.netbeans.modules.editor.plain/2
 	org.netbeans.modules.editor.plain.lib/1
 	org.netbeans.modules.editor.search
 	org.netbeans.modules.editor.settings/1
 	org.netbeans.modules.editor.settings.lib
 	org.netbeans.modules.editor.settings.storage/1
 	org.netbeans.modules.editor.structure/1
 	org.netbeans.modules.editor.tools.storage
 	org.netbeans.modules.editor.util/1
 	org.netbeans.modules.extexecution.impl
 	org.netbeans.modules.extexecution.process
 	org.netbeans.modules.extexecution.process.jdk9
 	org.netbeans.modules.favorites/1
 	org.netbeans.modules.git
 	org.netbeans.modules.gsf.testrunner/2
 	org.netbeans.modules.gsf.testrunner.ui
 	org.netbeans.modules.html/1
 	org.netbeans.modules.html.custom
 	org.netbeans.modules.html.editor/2
 	org.netbeans.modules.html.editor.lib/3
 	org.netbeans.modules.html.indexing
 	org.netbeans.modules.html.lexer/1
 	org.netbeans.modules.html.parser/1
 	org.netbeans.modules.html.validation/1
 	org.netbeans.modules.httpserver/2
 	org.netbeans.modules.hudson
 	org.netbeans.modules.hudson.git
 	org.netbeans.modules.hudson.mercurial
 	org.netbeans.modules.hudson.subversion
 	org.netbeans.modules.hudson.tasklist
 	org.netbeans.modules.hudson.ui
 	org.netbeans.modules.ide.kit
 	org.netbeans.modules.image/1
 	org.netbeans.modules.janitor
 	org.netbeans.modules.jumpto/1
 	org.netbeans.modules.keyring.fallback
 	org.netbeans.modules.keyring.impl
 	org.netbeans.modules.languages.diff
 	org.netbeans.modules.languages.manifest
 	org.netbeans.modules.languages.toml
 	org.netbeans.modules.languages.yaml
 	org.netbeans.modules.lexer/2
 	org.netbeans.modules.lexer.nbbridge/1
 	org.netbeans.modules.localhistory
 	org.netbeans.modules.localtasks
 	org.netbeans.modules.lsp.client/0
 	org.netbeans.modules.markdown
 	org.netbeans.modules.masterfs.nio2
 	org.netbeans.modules.mercurial
 	org.netbeans.modules.mylyn.util
 	org.netbeans.modules.nativeimage.api/0
 	org.netbeans.modules.netbinox
 	org.netbeans.modules.notifications
 	org.netbeans.modules.options.editor/1
 	org.netbeans.modules.parsing.api/1
 	org.netbeans.modules.parsing.indexing
 	org.netbeans.modules.parsing.lucene/2
 	org.netbeans.modules.parsing.nb
 	org.netbeans.modules.parsing.ui
 	org.netbeans.modules.print
 	org.netbeans.modules.print.editor
 	org.netbeans.modules.project.ant/1
 	org.netbeans.modules.projectapi/1
 	org.netbeans.modules.projectapi.nb
 	org.netbeans.modules.project.indexingbridge
 	org.netbeans.modules.project.libraries/1
 	org.netbeans.modules.project.libraries.ui/1
 	org.netbeans.modules.project.spi.intern
 	org.netbeans.modules.project.spi.intern.impl
 	org.netbeans.modules.projectui
 	org.netbeans.modules.projectuiapi/1
 	org.netbeans.modules.projectuiapi.base/1
 	org.netbeans.modules.properties/1
 	org.netbeans.modules.properties.syntax/1
 	org.netbeans.modules.refactoring.api
 	org.netbeans.modules.selenium2
 	org.netbeans.modules.selenium2.server
 	org.netbeans.modules.sendopts/2
 	org.netbeans.modules.server/0
 	org.netbeans.modules.servletapi/1
 	org.netbeans.modules.spellchecker
 	org.netbeans.modules.spellchecker.apimodule
 	org.netbeans.modules.spellchecker.bindings.htmlxml
 	org.netbeans.modules.spellchecker.bindings.properties
 	org.netbeans.modules.spellchecker.dictionary_en
 	org.netbeans.modules.spellchecker.kit
 	org.netbeans.modules.subversion
 	org.netbeans.modules.tasklist.kit
 	org.netbeans.modules.tasklist.projectint/1
 	org.netbeans.modules.tasklist.todo/1
 	org.netbeans.modules.tasklist.ui/1
 	org.netbeans.modules.team.commons
 	org.netbeans.modules.team.ide
 	org.netbeans.modules.templates/1
 	org.netbeans.modules.terminal
 	org.netbeans.modules.terminal.nb
 	org.netbeans.modules.textmate.lexer/0
 	org.netbeans.modules.usersguide/1
 	org.netbeans.modules.utilities/1
 	org.netbeans.modules.utilities.project/1
 	org.netbeans.modules.versioning/1
 	org.netbeans.modules.versioning.core/1
 	org.netbeans.modules.versioning.indexingbridge/0
 	org.netbeans.modules.versioning.masterfs
 	org.netbeans.modules.versioning.system.cvss.installer
 	org.netbeans.modules.versioning.ui/1
 	org.netbeans.modules.versioning.util
 	org.netbeans.modules.web.browser.api
 	org.netbeans.modules.web.common
 	org.netbeans.modules.web.common.ui
 	org.netbeans.modules.web.indent
 	org.netbeans.modules.xml
 	org.netbeans.modules.xml.axi
 	org.netbeans.modules.xml.catalog/2
 	org.netbeans.modules.xml.catalog.ui
 	org.netbeans.modules.xml.core/2
 	org.netbeans.modules.xml.lexer
 	org.netbeans.modules.xml.retriever/1
 	org.netbeans.modules.xml.schema.completion
 	org.netbeans.modules.xml.schema.model/1
 	org.netbeans.modules.xml.tax/2
 	org.netbeans.modules.xml.text/2
 	org.netbeans.modules.xml.tools/2
 	org.netbeans.modules.xml.xam/1
 	org.netbeans.modules.xml.xdm/1
 	org.netbeans.modules.xsl/1
 	org.netbeans.spi.editor.hints/0
 	org.netbeans.spi.editor.hints.projects
 	org.netbeans.spi.navigator/1
 	org.netbeans.spi.palette/1
 	org.netbeans.spi.tasklist/1
 	org.netbeans.swing.dirchooser
 	org.netbeans.swing.laf.dark/1
 	org.netbeans.swing.laf.flatlaf
 	org.openide.util.ui.svg
 	slf4j.api
 	slf4j.jdk14
</pre>
</details>

### Plugin Manager

This is most likely also reflected in the plugin manager. The Plugin-Manager in ThinkingRock (Menu `Tools` -> `Plugins` -> tab `Installed`) in the released version of ThinkingRock 3.7.0 only lists TR modules plus the library module `ext - commons-email`.
![image](https://user-images.githubusercontent.com/357238/212462286-12fb8882-2963-49d9-a1d3-3ecc7027e021.png)

In contrast, the development version lists many internal and NetBeans specific modules that should not be listed in the PluginManager:

![image](https://user-images.githubusercontent.com/357238/212463320-d7825f45-d1b0-402d-b1ca-26e4936e16bd.png)

We have to investigate the mechanics of such modules and how to prevent them from loading and/or showing in the plugin manager.

### Out of scope

A follow-up task not tackled in the scope of this ticket: Modules can be auto-updated from a central ThinkingRock specific update-center. Version management for modules (the version of the modules, not the dependencies) will be very important once we start publishing updated modules to that update center. So far I have not paid any attention to this. This can be tackled in the future with ticket #58.

**Comment by ursjoss** [comment]
By removing cluster `ide`, most of the excess plugins are gone. I also deactivated some modules from the `platform` cluster (e.g. test libraries), thus decreasing the size of the installer and thus the installed set of files. Both topics should be revisited in #44.

**Comment by RadekCap** [comment]
I would close this as fixed by #57.

**Comment by ursjoss** [comment]
The solution presented here did fix the issue reported, but it also had an ugly regression: It broke the build (see #98) without our CI being sensitive to this kind of breakage and report it with a failing build. (That will be improved in the context of #99).

With ticket #98, we revert the PR from this ticket - therefore I'm re-opening this ticket to be solved w/o the regression after #98 and #99 have been resolved.

**Comment by RadekCap** [comment]
Current status after #112 

This is the error message, after a first application run, tested on MacOS. There are no detected issues during usage of application after clicking "disable modules" during the first run.

`Warning - could not install some modules: Diff - The module named org.netbeans.modules.editor.fold/1 was needed and not found. Diff - The module named org.netbeans.modules.editor.settings/1 was needed and not found. Diff - The module named org.netbeans.modules.editor.lib2/1 was needed and not found. Diff - The module named org.netbeans.modules.options.editor/1 was needed and not found. Diff - The module named org.netbeans.modules.editor.util/1 was needed and not found. Diff - The module named org.netbeans.modules.editor.document was needed and not found. Diff - The module named org.netbeans.modules.editor.errorstripe/2 was needed and not found. Diff - The module named org.netbeans.modules.editor.lib/3 was needed and not found. Diff - The module named org.netbeans.modules.projectapi/1 was needed and not found. Multi-tabs project bridge - The module named org.netbeans.modules.projectuiapi.base/1 was needed and not found. Multi-tabs project bridge - The module named org.netbeans.modules.projectapi/1 was needed and not found.`

**Comment by ursjoss** [comment]
@RadekCap I couldn't really reproduce this on my machine. But hopfeully, PR #57 resolves this for you as well.

**Comment by RadekCap** [comment]
It's still there on both Mac and Linux. I'm attaching a screenshot from Linux. 

Could it be the cache? On Linux I have it broken and it couldn't run from the ZIP package because it still uses java 11. When it runs from Netbeans, it's ok. I removed .cache/trgtd/tr-4.0 and the issue still persists. Is there any other place where something is cached?

![Screenshot from 2023-03-06 09-22-12](https://user-images.githubusercontent.com/6508800/223056605-f35f5994-6061-4d4f-8404-1cd100abc8d8.png)

On my machines java is different - Temurin on Mac, RH build on Linux. Netbeans is the same - 17.
